### PR TITLE
revisited cmake MOD-4322

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,84 @@
+cmake_minimum_required(VERSION 3.10)
```
Comment (chayim) [inline CMakeLists.txt:1]: If you're basic the rest on 3.24, might as well just bump this to that IMHO.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+	set(CMAKE_CXX_FLAGS "-Wno-unused-parameter")
+
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fexceptions -fPIC ${CLANG_SAN_FLAGS} ${LLVM_CXX_FLAGS} ${COV_CXX_FLAGS}")
+	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} ${LLVM_LD_FLAGS}")
+	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} ${LLVM_LD_FLAGS}")
+
+	FetchContent_Declare(
+	googletest
+	URL https://github.com/google/googletest/archive/refs/tags/release-1.12.1.zip
```
Comment (chayim) [inline CMakeLists.txt:53]: variable for 1.12.1 that cmake can use?

File: Makefile [hunk 3]
```diff
@@ -107,14 +107,9 @@ else
...
 
-CMAKE_DIR=$(ROOT)/src
-
-CMAKE_FILES= \
-	src/CMakeLists.txt \
-	src/VecSim/spaces/CMakeLists.txt \
-	cmake/common.cmake \
-	cmake/gtest.cmake \
-	cmake/clang-sanitizers.cmake
+CMAKE_DIR=$(ROOT)
```
Comment (chayim) [inline Makefile:110]: CMAKE_DIR=$(ROOT)
CMAKE_TEST_DIR=$(ROOT)/tests
CMAKE_UNIT_TESTS=$(CMAKE_TEST_DIR)/unit
CMAKE_BENCHMARK_DIR=$(CMAKE_TEST_DIR)/benchmark

File: src/VecSim/spaces/CMakeLists.txt [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
 	IP/IP.cpp
 	)
 
-project(VectorSimilarity_Spaces)
 include(CheckCXXCompilerFlag)
 
-
+project(VectorSimilarity_Spaces)
 # TODO: Remove this once cpu_features get support for M1
 if(NOT APPLE)
```
Comment (chayim) [inline src/VecSim/spaces/CMakeLists.txt:13]: Wouldn't this be different on x86 apple?

File: src/VecSim/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+file(GLOB_RECURSE headers ./**.h)
+set(HEADER_LIST "${headers}")
+
+include_directories(../)
+
+add_subdirectory(spaces)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall")
+add_library(VectorSimilarity ${VECSIM_LIBTYPE}
+	algorithms/brute_force/brute_force_factory.cpp
```
Comment (chayim) [inline src/VecSim/CMakeLists.txt:16]: linting? spacing is a bit off

**Review by chayim — CHANGES_REQUESTED** [review]
small changes

File: Makefile (hunk 3 above)
Comment (DvirDukhan) [reply Makefile:110]: removed, not in use

File: src/VecSim/spaces/CMakeLists.txt (hunk 4 above)
Comment (DvirDukhan) [reply src/VecSim/spaces/CMakeLists.txt:13]: why?

**Pull request merged**